feat: showMore message add event info

zht826 opened the pull request:
![image](https://github.com/jquense/react-big-calendar/assets/21307445/0f88f7ed-2f8b-4e4b-85cb-2781fe3cdf76)
Sometimes, we need to incorporate additional event information into texts that display more items. Therefore, we have added the input of event parameters.

cutterbl commented:
Can you give me a practical use case for this?

zht826 commented:
such as this
![image](https://github.com/jquense/react-big-calendar/assets/21307445/6704d139-f03a-403f-9dcf-44c82c57a418)
I want to show the title of the remaining items

cutterbl commented:
Your code changes look good, but you should update the [messages](https://jquense.github.io/react-big-calendar/examples/index.html?path=/docs/props--messages) documentation prior to approval

zht826 commented:
I improved the parameters and added the remaining events and all-day event parameters, which correspond to the previous usage scenario.
I have updated the messages doc.
